hbase-server/src/test/java/org/apache/hadoop/hbase/security/provider/TestCustomSaslAuthenticationProvider.java

```diff
@@ -431,9 +431,13 @@ public static void setupCluster() throws Exception {
         UTIL.getDataTestDir("keytab").toUri().getPath());
     final MiniKdc kdc = UTIL.setupMiniKdc(KEYTAB_FILE);
 
-    // Switch back to NIO for now.
+    // Switch to blocking RPC impl.
     CONF.set(RpcClientFactory.CUSTOM_RPC_CLIENT_IMPL_CONF_KEY, BlockingRpcClient.class.getName());
     CONF.set(RpcServerFactory.CUSTOM_RPC_SERVER_IMPL_CONF_KEY, SimpleRpcServer.class.getName());
+    // Set the connection registry to ZKConnectionRegistry since hedging is not supported on
+    // blocking rpc clients.
```

Comment:
1. Why is this test forced onto the blocking rpc client? Can/Should it be parameterized over both implementations?
1. can we not use the `MasterRegistry` with hedged rpcs disabled? It appears this configuration is supported by the `MasterRegistry` constructor.

Reply:
> Why is this test forced onto the blocking rpc client? Can/Should it be parameterized over both implementations?

Not totally sure why it should only be a blocking rpc client (and a SimpleRpcServer impl). @joshelser Any idea? I see that you are the author of this test.

> can we not use the MasterRegistry with hedged rpcs disabled? It appears this configuration is supported by the MasterRegistry constructor.

We can. But my comment above has nothing to do with whether hedging is enabled/disabled. HedgingRpcChannel is a non-blocking channel implementation. Using a non-blocking-channel wrapped over a blocking-rpc-connection defeats the purpose.

Reply:
> Why is this test forced onto the blocking rpc client? Can/Should it be parameterized over both implementations?

Oversight on my part! Like Nick says -- we should use both via parameterization. There's another test which does this .... TestSecureIPC! 

Do you want to add that parameterization as a part of this PR? Or I can open up an issue and fix this test (as the negligent author ;)). LMK!

Reply:
No worries, I parameterized it. Thanks for the quick response.